crypto/include/internal/modes_int.h

```diff
@@ -1,14 +1,220 @@
 /*
- * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2010-2019 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
 
+#include <openssl/modes.h>
+
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
```

Comment:
Wait, you're moving stuff that's highly private for the modes sub-system (for lack of a better terms) to a header that's shareable to all sus-systems within libcrypto... That's not very sound, structurally speaking.

The reason I react is that we've had a *mess* in pre-1.1.0 openssl, where different subsystems were reaching into each other's internals all over the place, I would rather not see that being repeated. 

I am mostly away from keyboard this weekend, so I would like get a moment on Monday to look more closely at possibilities.

Reply:
Before this change the header file changes were being duplicated in multiple source files (this is not ideal either). So I am hoping that e_aes.c will eventually not need this header, and then the header is only needed by the provider aes files (yes I am seperating them into multiple files so it is not so messy to find things).
I would prefer that this PR doesnt get blocked for too long so that I can fix up all the other cipher related PR's that are dependent on this one.
Maybe the short term solution is a TODO? Long term the header should just be local in the providers.

Reply:
May I suggest a separate file, the same way you did with `aes_platform.h`, marked with a TODO saying that it will eventually be moved to the providers?

Reply:
So you just want me to add a todo to the existing modes_int.h - or you want part of the file seperated out?

Reply:
I want what you copied from `modes_locl.h` in a separate file, with a TODO.

Reply:
You realize everyting except for the siv stuff at the bottom was in modes_locl.h
I am not sure what seperating that small bit gives you?

Reply:
The stuff in `modes_lcl.h` should be private to the modes code only, i.e. only be included by `crypto/modes/*.c`, while the stuff in `modes_int.h` is available to all of libcrypto.  That's the basic principle, and moving it all of `modes_lcl.h` to `modes_int.h` breaks that separation.

Now, I know that some units include `modes_lcl.h`, and I see that as an abomination.  Either *some* stuff in `modes_lcl.h` can be made more general and moved to `modes_int.h`, or there is a fairly ugly hackery going on in `crypto/evp/`.  Considering how the principles behind coding the EVP backends have changed over the last few years, I suspect the latter, and probably fairly old hackery that was OK then and has remained because other priorities.

However, I would rather not confound the code further by cementing the ugly hackery.

Reply:
I have just moved the siv stuff that was in modes_int.h into a seperate file siv.h, and added a TODO to the modes_int.h

Reply:
[sigh] I'm not exactly happy with this solution, but ok, but please fix the comment nit I'm about to throw your way